src/Commands/NewCommand.php

```diff
@@ -18,187 +42,594 @@
-    private $client;
 
     /**
-     * @param ClientInterface $client
+     * @var OutputInterface
      */
-    function __construct(ClientInterface $client)
+    private $output;
+
+    protected function configure()
     {
-        $this->client = $client;
-        parent::__construct();
+        $this
+            ->setName('new')
+            ->setDescription('Creates a new ProcessWire project.')
+            ->addArgument('directory', InputArgument::REQUIRED, 'Directory where the new project will be created.')
+            ->addOption('dbUser', null, InputOption::VALUE_REQUIRED, 'Database user.')
+            ->addOption('dbName', null, InputOption::VALUE_REQUIRED, 'Database name.')
+            ->addOption('dbPass', null, InputOption::VALUE_REQUIRED, 'Database password.')
+            ->addOption('dbHost', null, InputOption::VALUE_REQUIRED, 'Database host.')
+            ->addOption('dbPort', null, InputOption::VALUE_REQUIRED, 'Database port.')
+            ->addOption('dbEngine', null, InputOption::VALUE_REQUIRED, 'Database engine.')
+            ->addOption('dbCharset', null, InputOption::VALUE_REQUIRED, 'Database characterset.')
+            ->addOption('timezone', null, InputOption::VALUE_REQUIRED, 'Timezone.')
+            ->addOption('chmodDir', null, InputOption::VALUE_REQUIRED, 'Directory mode. Defaults 755.')
+            ->addOption('chmodFile', null, InputOption::VALUE_REQUIRED, 'File mode. Defaults 644')
+            ->addOption('httpHosts', null, InputOption::VALUE_REQUIRED, 'Hostname without www part.')
+            ->addOption('adminUrl', null, InputOption::VALUE_REQUIRED, 'Admin url.')
+            ->addOption('username', null, InputOption::VALUE_REQUIRED, 'Admin username.')
+            ->addOption('userpass', null, InputOption::VALUE_REQUIRED, 'Admin password.')
+            ->addOption('useremail', null, InputOption::VALUE_REQUIRED, 'Admin email address.')
+            ->addOption('profile', null, InputOption::VALUE_REQUIRED, 'Default site profile.')
+            // ->addArgument('color', null, InputOption::VALUE_NONE, 'Default ProcessWire color.')
+            ->addOption('dev', null, InputOption::VALUE_NONE, 'Download dev branch')
+            ->addOption('no-install', null, InputOption::VALUE_NONE, 'Disable installation');
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->fs = new Filesystem();
+        $directory = rtrim(trim($input->getArgument('directory')), DIRECTORY_SEPARATOR);
+        $this->projectDir = $this->fs->isAbsolutePath($directory) ? $directory : getcwd().DIRECTORY_SEPARATOR.$directory;
+        $this->projectName = basename($directory);
+
+        $logger = new Logger('name');
+        $logger->pushHandler(new StreamHandler("php://output"));
+        $this->installer = new Installer($logger, $this->projectDir);
+
+        // $this->version = trim($input->getArgument('version'));
```

Comment:
Hi @marcus-herrmann , the idea is at some point of time when PW add tags to all release, we can pass and get the version. I like it to stay there, if you are very strong on it, will remove the same when I am back.